grid.clip() method incompatible with rasterio 1.4.0

The following error is returned with this test:
```
tests/test_grid.py:57: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <interpies.grid.Grid object at 0x7fb153933f50>, xmin = 1500
xmax = 2000.5, ymin = 0, ymax = 1000

    def clip(self, xmin, xmax, ymin, ymax) -> 'Grid':
        """
        Clip grid.
        """
        rows, cols = rasterio.transform.rowcol(self.transform, [xmin, xmax], [ymin, ymax])
        rows = np.clip(rows, 0, None)
        cols = np.clip(cols, 0, None)
>       data_selection = self.data[rows[1] : rows[0] + 1, cols[0] : cols[1] + 1]
E       TypeError: slice indices must be integers or None or have an __index__ method
```

--- a/interpies/grid.py
+++ b/interpies/grid.py
@@ -137,8 +137,8 @@ class Grid:
         Clip grid.
         """
         rows, cols = rasterio.transform.rowcol(self.transform, [xmin, xmax], [ymin, ymax])
-        rows = np.clip(rows, 0, None)
-        cols = np.clip(cols, 0, None)
+        rows = np.asarray(np.clip(rows, 0, None), dtype=np.int64)
+        cols = np.asarray(np.clip(cols, 0, None), dtype=np.int64)
         data_selection = self.data[rows[1] : rows[0] + 1, cols[0] : cols[1] + 1]
 
         # new origin

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -25,7 +25,7 @@ dependencies = [
     "matplotlib",
     "numpy",
     "gdal==3.8.4",
-    "rasterio>=1.3,<1.4",
+    "rasterio>=1.3",
     "scikit-image",
     "scikit-learn",
     "scipy",

